### PR TITLE
install: fix bug #1823

### LINKED ISSUE #1823: install: does not remove path from files
While compiling python3 in nixpkgs I noticed that install does not properly work:

Python uses the similar commands to the following command autogenerated by autoconf to install files:
```
/store/7rj52iy8gvdwkrca5k35a7i4hhdgf90c-uutils-coreutils-0.0.4/bin/install -c -m 644 ./Lib/unittest/runner.py /nix/store/bg19k3h973swd0745mhhp5snzijz7hx3-python3-3.8.8/lib/python3.8/unittest
```

but install converts that to a wrong path and then prints that the directory does not exist:

```
install: error: cannot install './Lib/unittest/runner.py' to '/nix/store/bg19k3h973swd0745mhhp5snzijz7hx3-python3-3.8.8/lib/python3.8/unittest/./Lib/unittest/runner.py': the source path is not an existing regular file
```

A minimal reproduce is:

```
mkdir -p Lib/unittest lib/unittest
touch Lib/unittest/runner.py
install ./Lib/unittest/runner.py lib/unittest
install: error: cannot install './Lib/unittest/runner.py' to 'lib/unittest/./Lib/unittest/runner.py': No such file or directory (os error 2)
```

### PR DESCRIPTION
**Pull request by nomius10**
TIL path manipulation in rust is intentionally basic due to [security concerns](https://github.com/rust-lang/rust/issues/16507#issuecomment-73804876).

fixes #1823

### CODE REVIEW

File: src/uu/install/src/install.rs [hunk 1]
```diff
@@ -426,18 +426,25 @@ fn copy_files_into_dir(files: &[PathBuf], target_dir: &PathBuf, b: &Behavior) ->
...
-        let targetpath = match sourcepath.as_os_str().to_str() {
-            Some(name) => target_dir.join(name),
-            None => {
-                show_error!(
-                    "cannot stat '{}': No such file or directory",
-                    sourcepath.display()
-                );
+        if !sourcepath.exists() {
+            show_error!(
+                "cannot stat '{}': No such file or directory",
```
Comment (sylvestre) [inline src/uu/install/src/install.rs:431]: is it the same output as in GNU?

**Review by sylvestre — CHANGES_REQUESTED** [review]
Looks good but would it be great to add a test !

File: src/uu/install/src/install.rs (hunk 1 above)
Comment (nomius10) [reply src/uu/install/src/install.rs:431]: Yes

```
┌SLATEデnomiusガ14:58:57時install_bugfix=デ「~/P/c/bugtest」
└$ tree
.
├── Lib
│   └── unittest
│       ├── asd
│       │   └── stuff
│       └── runner.py
└── lib
    └── unittest
        └── runner.py

5 directories, 3 files
┌SLATEデnomiusガ14:59:00時install_bugfix=デ「~/P/c/bugtest」
└$ install ./Lib/unittest/{runner.py,asd} lib/unittest
install: omitting directory './Lib/unittest/asd'
```
(install is system install, not uutils)

Comment (nomius10) [reply src/uu/install/src/install.rs:431]: oh, I answered a different question.
This particular output is the same as before. although it should be `show_info!()` not `show_error!()`

Comment (nomius10) [reply src/uu/install/src/install.rs:431]: fixed

**Review by nomius10 — COMMENTED** [review]
For some reason I can't run any of the tests for `install`:
``` 
cargo test test_install_copy_file_leading_dot 
cargo test --bin=install --package=uu_install 
```

``` 
running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out 
```

I think that has to do with the fact that currently `install` is not included in the busybox-like binary

**Comment by sylvestre** [comment]
`cargo test --features feat_os_unix `
is a way to run tests

File: src/uu/install/src/install.rs [hunk 2]
```diff
@@ -426,18 +426,25 @@ fn copy_files_into_dir(files: &[PathBuf], target_dir: &PathBuf, b: &Behavior) ->
...
-            Some(name) => target_dir.join(name),
-            None => {
-                show_error!(
-                    "cannot stat '{}': No such file or directory",
-                    sourcepath.display()
-                );
+        if !sourcepath.exists() {
+            show_info!(
+                "cannot stat '{}': No such file or directory",
+                sourcepath.display()
```
Comment (sylvestre) [inline src/uu/install/src/install.rs:432]: Could you please make sure that the tests are exercising this code path?
thanks!

**Comment by nomius10** [comment]
I've snuck in a refactoring commit: I've renamed the file & directory names throughout the tests, since they gave a false impression that they all execute within the same directory (`at_and_ucmd!()` macro creates a new temporary directory for each individual test).

**Comment by sylvestre** [comment]
great, thanks :)

**Pull request merged**